src/main/java/pl/stormit/tinyurl/service/IpLocalizationService.java

```diff
@@ -0,0 +1,44 @@
+package pl.stormit.tinyurl.service;
+
+import com.maxmind.geoip2.DatabaseReader;
+import com.maxmind.geoip2.exception.AddressNotFoundException;
+import com.maxmind.geoip2.exception.GeoIp2Exception;
+import com.maxmind.geoip2.model.CityResponse;
+import org.springframework.stereotype.Service;
+import pl.stormit.tinyurl.dto.UrlAnalyticsLocalizationDto;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.InetAddress;
+
+@Service
+public class IpLocalizationService {
+
+    public static final String LOCALIZATION_DB_GEO_LITE = "src/main/resources/localizationdb/GeoLite2-City.mmdb";
+
+    public UrlAnalyticsLocalizationDto getIpLocalization(String addressIp) {
+
+        InetAddress ipAddress;
+        UrlAnalyticsLocalizationDto setLocalization = new UrlAnalyticsLocalizationDto();
+
+        try {
+            ipAddress = InetAddress.getByName(addressIp);
+
+            DatabaseReader database = new DatabaseReader.
+                    Builder(new File(LOCALIZATION_DB_GEO_LITE)).build();
+            CityResponse response = database.city(ipAddress);
+
+            if (response != null) {
+                setLocalization.setCountryLocalization(response.getCountry().getName());
```

Comment:
you can think what will happen if "getCountry" will return null.

Reply:
I changed the condition to check the result of the get method - so that there is no NPE.